android/build.gradle

```diff
@@ -22,3 +22,13 @@ allprojects {
         }
     }
 }
+
+subprojects {
+    afterEvaluate {project ->
+        if (project.hasProperty("android")) {
+            android {
+                buildToolsVersion '27.0.1'
```

Comment:
How did you choose the `subprojects` / `afterEvaluate` form for doing this?

From what I've read about Grade config I know there are several choices for how to make a patch like this, and they have subtly different meaning e.g. in what gets evaluated first, and what gets to override what. So I'd like to make sure we/I understand what choice we're making here.

Reply:
This version is adopted from https://docs.gradle.org/current/userguide/build_lifecycle.html#sec:project_evaluation

> subprojects {

Means only for subproject, we are adding this in the root project (which also contains our app as one of the project).

> afterEvaluate {project ->

This is a kinda callback (described as `closures` in the docs). Called as soon as project is evaluated, but before executing.

> if (project.hasProperty("android")) {

Making sure that we only override* variable value for android project/dependency, as `buildToolsVersion` is only required for android project.

* By doing this we are injecting/changing value after project is evaluating and waiting for executing (building project).


Another way is implementing interface listener (something like https://docs.gradle.org/current/javadoc/org/gradle/BuildListener.html).

Currently our gradle files follow `closures` approach only, so I thinks we should use `closures` only. 

like 
```
allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}
```

I completely agrees with the all of this 😄 , and in case if something went wrong we can definitely improve in future 😅 . For now I thinks this is sufficient 😄 .

(Now updating commit msg and rearranging)